
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=cp1252" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Heat Map</title>
    <link rel="stylesheet" type="text/css" href="../../_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/alabaster.css" />
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="next" title="Scatter Plot" href="scatterplot.html" />
    <link rel="prev" title="Distributions" href="distributions.html" />
   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="heat-map">
<h1>Heat Map</h1>
<p>Plots a heat map for a pair of attributes.</p>
<p><strong>Inputs</strong></p>
<ul class="simple">
<li>Data: input dataset</li>
</ul>
<p><strong>Outputs</strong></p>
<ul class="simple">
<li>Selected Data: instances selected from the plot</li>
<li>Data: data with an additional column showing whether a point is selected</li>
</ul>
<p><a class="reference external" href="https://en.wikipedia.org/wiki/Heat_map" target="_blank">Heat map</a> is a graphical method for visualizing attribute values in a two-way matrix. It only works on datasets containing numeric variables. The values are represented by color according to the selected color pallette. By combining class variable and attributes on x and y axes, we see where the attribute values are the strongest and where the weakest, thus enabling us to find typical features for each class.</p>
<p>The widget enables row selection with click and drag. One can zoom in with Ctrl++ (Cmd++) and zoom out with Ctrl+- (Cmd+-). Ctrl+0 (Cmd+0) resets zoom to the extended version, while Ctrl+9 (Cmd+9) reset it to the default.</p>
<p><img alt="../../_images/HeatMap.png" src="../../_images/HeatMap.png" /></p>
<ol class="simple">
<li>The color pallette. Choose from linear, diverging, color-blind friendly, or other pallettes. <strong>Low</strong> and <strong>High</strong> are thresholds for the color palette (low for attributes with low values and high for attributes with high values). Selecting one of diverging palettes, which have two extreme colors and a neutral (black or white) color at the midpoint, enables an option to set a meaningful mid-point value (default is 0).</li>
<li>Merge rows. If there are too many rows in the visualization, one can merge them with k-means algorithm into N selected clusters (default 50).</li>
<li>Cluster columns and rows:<ul>
<li><strong>None</strong> (lists attributes and rows as found in the dataset)</li>
<li><strong>Clustering</strong> (clusters data by similarity with hierarchical clustering on Euclidean distances and with average linkage)</li>
<li><strong>Clustering with ordered leaves</strong> (same as clustering, but it additionally maximizes the sum of similarities of adjacent elements)</li>
</ul>
</li>
<li>Split rows or columns by a categorical variable. If the data contains a class variable, rows will be automatically split by class.</li>
<li>Set what is displayed in the plot in <strong>Annotation &amp; Legend</strong>.<ul>
<li>If <em>Show legend</em> is ticked, a color chart will be displayed above the map.</li>
<li>If <em>Stripes with averages</em> is ticked, a new line with attribute averages will be displayed on the left.
<strong>Row Annotations</strong> adds annotations to each instance on the right. Color colors the instances with the corresponding value of the selected categorical variable.
<strong>Column Annotations</strong> adds annotation to each variable at the selected position (default is Top). Color colors the columns with the corresponding value of the selected column annotation.</li>
</ul>
</li>
<li>If <em>Keep aspect ratio</em> is ticked, each value will be displayed with a square (proportionate to the map).</li>
<li>If <em>Send Automatically</em> is ticked, changes are communicated automatically. Alternatively, click <em>Send</em>.</li>
</ol>
<div class="section" id="advanced-visualization">
<h2>Advanced visualization</h2>
<p>Heat map enables some neat plot enhancements. Such options are clustering of rows and/or columns for better data organization, row and column annotations, and splitting the data by categorical variables.</p>
<p>Row and column clustering is performed independently. Row clustering is computed from Euclidean distances, while column clustering uses Pearson correlation coefficients. Hierarchical clustering is based on the Ward linkage method. Clustering with optimal leaf ordering reorders left and right branches in the dendrogram to minimize the sum of distances between adjacent leaves (Bar-Joseph et al. 2001).</p>
<p><img alt="../../_images/HeatMap-advanced.png" src="../../_images/HeatMap-advanced.png" /></p>
</div>
<div class="section" id="examples">
<h2>Examples</h2>
<div class="section" id="gene-expresssions">
<h3>Gene expresssions</h3>
<p>The <strong>Heat Map</strong> below displays attribute values for the <em>brown-selected</em> data set (Brown et al. 2000). Heat maps are particularly appropriate for showing gene expressions and the brown-selected data set contains yeast gene expressions at different conditions.</p>
<p>Heat map shows low expressions in blue and high expressions in yellow and white. For better organization, we added <em>Clustering (opt. ordering)</em> to the columns, which puts columns with similar profiles closer together. In this way we can see the conditions that result in low expressions for ribosomal genes in the lower right corner.</p>
<p>Additionally, the plot is enhanced with row color on the right, showing which class the rows belong to.</p>
<p><img alt="../../_images/HeatMap-Example1.png" src="../../_images/HeatMap-Example1.png" /></p>
</div>
<div class="section" id="sentiment-analysis">
<h3>Sentiment Analysis</h3>
<p>Heat maps are great for visualizing any kind of comparable numeric variables, for example sentiment in a collection of documents. We will take <em>book-excerpts</em> corpus from the <strong>Corpus</strong> widget and pass it to the <strong>Sentiment Analysis</strong> widget, which computes sentiment scores for each document. The output of sentiment analysis are four columns, positive, negative, and neutral sentiment score, and a compound score that aggregates the previous scores into a single number. Positive compound values (white) represent positive documents, while negative (blue) represent negative documents.</p>
<p>We used row clustering to place similar rows closer together, resulting in clear negative and positive groups. Now we can select negative children’s books and explore which are they.</p>
<p><img alt="../../_images/HeatMap-Example2.png" src="../../_images/HeatMap-Example2.png" /></p>
</div>
</div>
<div class="section" id="references">
<h2>References</h2>
<p>Bar-Joseph, Z., Gifford, D.K., Jaakkola, T.S. (2001) Fast optimal leaf ordering for hierarchical clustering, Bioinformatics, 17, 22-29.</p>
<p>Brown, M.P., Grundy, W.N., Lin, D., Cristianini, N., Sugnet, C., Furey, T.S., Ares, M., Haussler, D. (2000) Knowledge-based analysis of microarray gene expression data by using support vector machines, Proceedings of the National Academy of Sciences, 1, 262-267.</p>
</div>
</div>


          </div>
          
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2015, Orange Data Mining.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.5.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../_sources/widgets/visualize/heatmap.md.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>